go/oasis-node/cmd/common/metrics/resource.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/spf13/viper"
+
+	"github.com/oasislabs/oasis-core/go/common/service"
+)
+
+// ResourceCollector is interface for monitoring resources (cpu, mem, disk, net).
+type ResourceCollector interface {
+	// Update updates corresponding resource metrics.
+	Update() error
+}
+
+// resourceService updates and collects number of resources.
+type resourceService struct {
+	service.BaseBackgroundService
+
+	interval           time.Duration
+	resourceCollectors []ResourceCollector
+}
+
+func (brs *resourceService) Start() error {
+	go brs.worker()
+	return nil
+}
+
+func (brs *resourceService) worker() {
+	t := time.NewTicker(brs.interval)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-brs.Quit():
+			break
+		case <-t.C:
+		}
+		for _, rs := range brs.resourceCollectors {
+			if err := rs.Update(); err != nil {
+				brs.Logger.Warn(err.Error())
```

Comment:
This is a bad idea as we use structured logging, do something like:
```suggestion
				brs.Logger.Warn("failed to update resource collector", "collector", rs.Name(), "err", err)
```

And consider adding a `Name() string` method to the resource collector interface.